src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -165,7 +165,9 @@ private static boolean isPublicStaticFinalIntField(DetailAST ast) {
         boolean result = ast.getType() == TokenTypes.VARIABLE_DEF;
         if (result) {
             final DetailAST type = ast.findFirstToken(TokenTypes.TYPE);
-            result = type.getFirstChild().getType() == TokenTypes.LITERAL_INT;
+            final DetailAST arrayDeclarator = type.getFirstChild().getNextSibling();
+            result = arrayDeclarator == null
```

Comment:
How come the changes to `FinalParameters` uses `findFirstToken(TokenTypes.ARRAY_DECLARATOR)` but this one checks child positioning?

Reply:
Probably laziness, it is likely that I just modified the existing code. I can change it, if you prefer one or the other. Getting the exact child that we want might be less "expensive" than traversing the children of a particular node a la `findFirstToken()`.